community/security/src/main/java/org/neo4j/server/security/auth/AbstractUserRepository.java

```diff
@@ -89,7 +89,8 @@ public void update( User existingUser, User updatedUser )
         // Assert input is ok
         if ( !existingUser.name().equals( updatedUser.name() ) )
         {
-            throw new InvalidArgumentsException( "updated user has a different name" );
+            throw new IllegalArgumentException( "Changing a user's name is not allowed. Change attempted from name '" +
+                    existingUser.name() + "' to '" + updatedUser.name() + "'." );
```

Comment:
This is definitely tricky! We need to bring in also the reason why this (changing a username) cannot happen.
So, I would suggest changing to, where XYZ is the reason why we do not allow a mutable username: 
"Changing a user's username is not permitted as XYZ. The attempt to change the username from 'existing' to 'updated' has failed".
I realise this is not something the user/admin would consciously have attempted to do, and that this situation only arises in very limited scenarios (resulting from flakiness?)